### PR TITLE
Allow anonymous to be configured as the username of consumer

### PR DESCRIPTION
**Pull request by vm-001**
### Summary

Allow anonymous to be configured as the username of the consumer.

### Testing

I've never tested yet

FTI-3109

https://github.com/Kong/kong/pull/9371

### CODE REVIEW

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

File: app/_hub/kong-inc/hmac-auth/_index.md [hunk 1]
```diff
@@ -86,8 +86,18 @@ params:
...
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
+      minimum_version: "3.1.0"
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string    
       description: |
-        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails. If empty (default), the request fails with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` attribute which is internal to Kong, and **not** its `custom_id`.
+        An optional string (consumer UUID) value to use as an anonymous Consumer if authentication fails.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/hmac-auth/_index.md:97]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
```

File: app/_hub/kong-inc/hmac-auth/_index.md [hunk 2]
```diff
@@ -86,8 +86,18 @@ params:
...
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/hmac-auth/_index.md:99]:
```suggestion
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/jwt/_index.md [hunk 3]
```diff
@@ -113,8 +113,18 @@ params:
       required: false
       default: null
       datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/jwt/_index.md:117]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/jwt/_index.md [hunk 4]
```diff
@@ -113,8 +113,18 @@ params:
...
+      minimum_version: "3.1.0"
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string      
       description: |
-        An optional string (consumer uuid) value to use as an anonymous consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. The anonymous value must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
+        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/jwt/_index.md:124]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
```

File: app/_hub/kong-inc/jwt/_index.md [hunk 5]
```diff
@@ -113,8 +113,18 @@ params:
...
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/jwt/_index.md:126]:
```suggestion
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/key-auth-enc/_index.md [hunk 6]
```diff
@@ -80,14 +80,22 @@ params:
       description: |
         An optional boolean value telling the plugin to show or hide the credential from the upstream service. If `true`,
         the plugin strips the credential from the request (i.e., the header, query string, or request body containing the key) before proxying it.
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth-enc/_index.md:88]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/key-auth-enc/_index.md [hunk 7]
```diff
@@ -80,14 +80,22 @@ params:
...
       required: false
       default: null
       datatype: string
       description: |
-        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails. If empty (default),
-        the request will fail with an authentication failure `4xx`. Note that this value
-        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
+        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth-enc/_index.md:97]:
```suggestion
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/key-auth/_index.md [hunk 8]
```diff
@@ -112,14 +112,22 @@ params:
       description: |
         An optional boolean value telling the plugin to show or hide the credential from the upstream service. If `true`,
         the plugin strips the credential from the request (i.e., the header, query string, or request body containing the key) before proxying it.
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth/_index.md:120]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/key-auth/_index.md [hunk 9]
```diff
@@ -112,14 +112,22 @@ params:
...
     - name: anonymous
       required: false
       default: null
       datatype: string
       description: |
-        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails.
+        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
         If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
-        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
+        must refer to the consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/key-auth/_index.md:129]:
```suggestion
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth-advanced/_index.md [hunk 10]
```diff
@@ -127,16 +127,24 @@ params:
       datatype: number
       description: |
         An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
+    - name: anonymous
+      required: false
+      default: null
+      value_in_examples: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/ldap-auth-advanced/_index.md:136]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth-advanced/_index.md [hunk 11]
```diff
@@ -127,16 +127,24 @@ params:
...
       value_in_examples: null
       datatype: string
       description: |
-        An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the
-        request will fail with an authentication failure `4xx`.
-
-        **Note:** The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
+        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/ldap-auth-advanced/_index.md:146]:
```suggestion
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth/_index.md [hunk 12]
```diff
@@ -137,14 +137,22 @@ params:
       datatype: number
       description: |
         An optional value in milliseconds that defines how long an idle connection to LDAP server will live before being closed.
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/ldap-auth/_index.md:145]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/ldap-auth/_index.md [hunk 13]
```diff
@@ -137,14 +137,22 @@ params:
...
       required: false
       default: null
       datatype: string
       description: |
-        An optional string (consumer UUID) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`.
-
-        **Note:** The value must refer to the Consumer `id` attribute that is internal to Kong, **not** its `custom_id`.
+        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/ldap-auth/_index.md:154]:
```suggestion
        If empty (default), the request fails with an authentication failure `4xx`. Note that this value
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/mtls-auth/_index.md [hunk 14]
```diff
@@ -39,13 +39,22 @@ params:
   config:
     - name: anonymous
       required: false
+      default: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/mtls-auth/_index.md:45]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/oauth2-introspection/_index.md [hunk 15]
```diff
@@ -102,13 +102,22 @@ params:
       datatype: integer
       description: |
         An optional value in milliseconds that defines how long an idle connection lives before being closed.
+    - name: anonymous
+      required: false
+      default: null
+      value_in_examples: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/oauth2-introspection/_index.md:111]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/oauth2-introspection/_index.md [hunk 16]
```diff
@@ -102,13 +102,22 @@ params:
...
+      minimum_version: "3.1.0"
     - name: anonymous
       required: false
       default: null
       value_in_examples: null
       datatype: string
       description: |
-        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure 4xx.
+        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails. If empty (default), the request will fail with an authentication failure 4xx.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/oauth2-introspection/_index.md:119]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails. If empty (default), the request fails with an authentication failure `4xx`.
```

File: app/_hub/kong-inc/oauth2/_index.md [hunk 17]
```diff
@@ -142,8 +142,18 @@ params:
       required: false
       default: null
       datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/oauth2/_index.md:146]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/oauth2/_index.md [hunk 18]
```diff
@@ -142,8 +142,18 @@ params:
...
+      minimum_version: "3.1.0"
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string      
       description: |
-        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` attribute which is internal to Kong, and **not** its `custom_id`.
+        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/oauth2/_index.md:153]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
```

File: app/_hub/kong-inc/oauth2/_index.md [hunk 19]
```diff
@@ -142,8 +142,18 @@ params:
...
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/oauth2/_index.md:155]:
```suggestion
        If empty (default), the request fails with an authentication failure `4xx`. Note that this value
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 20]
```diff
@@ -178,10 +178,19 @@ params:
     - name: anonymous
       required: false
       default: null
-      datatype: uuid
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/openid-connect/_index.md:183]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 21]
```diff
@@ -178,10 +178,19 @@ params:
...
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string      
       description: |
-        Let unauthenticated requests pass or skip the plugin if another authentication plugin
-        has already authenticated the request by setting the value to anonymous Consumer.
+        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails.
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/openid-connect/_index.md:192]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/vault-auth/_index.md [hunk 22]
```diff
@@ -54,14 +54,22 @@ params:
       datatype: boolean
       description: |
         An optional boolean value telling the plugin to show or hide the credential from the upstream service. If `true`, the plugin will strip the credential from the request (i.e. the header or querystring containing the key) before proxying it.
+    - name: anonymous
+      required: false
+      default: null
+      datatype: string
+      description:
+        An optional string (Consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request will fail with an authentication failure `4xx`. Please note that this value must refer to the Consumer `id` or `username` attribute, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/vault-auth/_index.md:62]:
```suggestion
        An optional string (consumer UUID or username) value to use as an “anonymous” consumer if authentication fails. If empty (default null), the request fails with an authentication failure `4xx`. Note that this value must refer to the consumer `id` or `username` attribute, and **not** its `custom_id`.
```

File: app/_hub/kong-inc/vault-auth/_index.md [hunk 23]
```diff
@@ -54,14 +54,22 @@ params:
...
       required: false
       default: null
       datatype: string
       description: |
-        An optional string (consumer uuid) value to use as an "anonymous" consumer if authentication fails. If empty (default), the request will fail with an authentication failure `4xx`.
-
-        **Note:** This value must refer to the Consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
+        An optional string (Consumer UUID) value to use as an anonymous Consumer if authentication fails.
+        If empty (default), the request will fail with an authentication failure `4xx`. Note that this value
+        must refer to the Consumer `id` attribute that is internal to Kong, and **not** its `custom_id`.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/vault-auth/_index.md:71]:
```suggestion
        An optional string (consumer UUID) value to use as an anonymous consumer if authentication fails.
        If empty (default), the request fails with an authentication failure `4xx`. Note that this value
        must refer to the consumer `id` attribute that is internal to Kong Gateway, and **not** its `custom_id`.
```

**Review by cloudjumpercat — APPROVED** [review]
LGTM!

**Comment by cloudjumpercat** [comment]
Ignoring Vale errors because they are flagging Liquid syntax incorrectly.

**Comment by cloudjumpercat** [comment]
@vm-001 Should this wait for the 3.1.x release to be merged?

**Comment by vm-001** [comment]
@cloudjumpercat Thanks! Yes, this PR should wait for 3.1.x release, that's why I add the 3.1 Milestone.

**Comment by cloudjumpercat** [comment]
Ignoring Vale errors, those will be resolved when this is merged into main (because the Vale plugin changes aren't on this branch).

**Pull request merged**